webhook/README.md

```diff
@@ -2,73 +2,52 @@ Webhook
+You can get all full data of request webhook from variable `request`
 Example with github:
-```python
-@api.one
-def set_event(self):
-    super(Webhook, self).set_event()
-    if not self.env.webhook_event and self.env.webhook_driver_name == 'github':
-        self.env.webhook_event = self.env.request.httprequest.headers.get('X-Github-Event')
+```xml
+<!--webhook github data-->
+<record model="webhook" id="webhook_github">
+    <field name="name">github</field>
+    <field name="python_code_get_event">request.httprequest.headers.get('X-Github-Event')</field>
+</record>
 ```
 
-
-Add new methods named `"run_webhook_" + consumer_name + "_" + consumer_event` auto invoke in each webhook request.
-Example with githup:
-```python
-@api.one
-def run_webhook_github_push(self):
-    # logic to run webhook event called "push" of "github"
-    sys.stdout.write("Push event in: %s/%s\n" % (
-        self.env.request.jsonrequest['repository']['owner']['name'],
-        self.env.request.jsonrequest['repository']['name']
-    ))
-```
-
-You can access to full request webhook data in variable `self.env.request`
-
-Complete example with github:
-```python
-NAME_OF_YOUR_WEBHOOK_CONSUMER='github'
-IP_OR_SUBNET_YOUR_WEBHOOK = ['192.30.252.0/22'] # https://help.github.com/articles/what-ip-addresses-does-github-use-that-i-should-whitelist/#current-ip-addresses
-
-class Webhook(models.Model):
-    _inherit = 'webhook'
-
-    @api.one
-    def set_driver_remote_address(self):
-        super(Webhook, self).set_driver_remote_address()
-        self.env.webhook_driver_address.update({
-            NAME_OF_YOUR_WEBHOOK_CONSUMER: IP_OR_SUBNET_YOUR_WEBHOOK
-        })
-
-    @api.one
-    def set_event(self):
-        super(Webhook, self).set_event()
-        if not self.env.webhook_event and \
-                self.env.webhook_driver_name == NAME_OF_YOUR_WEBHOOK_CONSUMER:
-            self.env.webhook_event = self.env.request.httprequest.headers.get('X-Github-Event')
-
-    @api.one
-    def run_webhook_github_push(self):
-        # logic to run webhook event called "push" of "github"
-        sys.stdout.write("Push event in: %s/%s\n" % (
-            self.env.request.jsonrequest['repository']['owner']['name'],
-            self.env.request.jsonrequest['repository']['name']
-        ))
+Full example of create a new webhook configuration data.
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
+
+        <!--webhook github data-->
+        <record model="webhook" id="webhook_github">
+            <field name="name">github</field>
+            <field name="python_code_get_event">request.httprequest.headers.get('X-Github-Event')</field>
+        </record>
+        <!--webhook github data of remote address-->
+        <record model="webhook.address" id="webhook_address_github">
+            <field name="name">192.30.252.0/22</field>
```

Comment:
github has this [IP's](https://help.github.com/articles/what-ip-addresses-does-github-use-that-i-should-whitelist/)

Token matter [here](https://github.com/Vauxoo/server-tools/issues/9)